### PR TITLE
DO NOT MERGE: Test Pronto

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

File: lib/gooddata/smelly.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Smelly < String
+  @@class_variable = :whatever
+  attr_accessor :dummy
```
Comment (yenkins) [inline lib/gooddata/smelly.rb]: Is a writable attribute (Attribute)

File: lib/gooddata/smelly.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Smelly < String
+  @@class_variable = :whatever
```
Comment (yenkins) [inline lib/gooddata/smelly.rb]: Declares the class variable '@@class_variable' (ClassVariable)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Style/ClassVars: Replace class var @@class_variable with a class instance var.

File: lib/gooddata/smelly.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  attr_accessor :dummy
+
+  def x(y1,y2); end
```
Comment (yenkins) [inline lib/gooddata/smelly.rb]: Takes parameters ['y1', 'y2'] to 3 methods (DataClump)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has the name 'x' (UncommunicativeMethodName)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has the parameter name 'y1' (UncommunicativeParameterName)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has the parameter name 'y2' (UncommunicativeParameterName)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has unused parameter 'y1' (UnusedParameters)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has unused parameter 'y2' (UnusedParameters)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Style/SpaceAfterComma: Space missing after comma.

File: lib/gooddata/smelly.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class Smelly < String
```
Comment (yenkins) [inline lib/gooddata/smelly.rb]: Assumes too much for instance variable '@ivar' (InstanceVariableAssumption)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has no descriptive comment (IrresponsibleModule)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has prima donna method 'bar!' (PrimaDonnaMethod)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Inherits from core class 'string' (SubclassedFromCoreClass)

File: lib/gooddata/smelly.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  @@class_variable = :whatever
+  attr_accessor :dummy
+
+  def x(y1,y2); end
+  def y(y1,y2); end
```
Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has the name 'y' (UncommunicativeMethodName)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has the parameter name 'y1' (UncommunicativeParameterName)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has the parameter name 'y2' (UncommunicativeParameterName)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has unused parameter 'y1' (UnusedParameters)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has unused parameter 'y2' (UnusedParameters)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Style/EmptyLineBetweenDefs: Use empty lines between method definitions.

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Style/SpaceAfterComma: Space missing after comma.

File: lib/gooddata/smelly.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  def z(y1,y2); end
```
Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has the name 'z' (UncommunicativeMethodName)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has the parameter name 'y1' (UncommunicativeParameterName)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has the parameter name 'y2' (UncommunicativeParameterName)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has unused parameter 'y1' (UnusedParameters)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Has unused parameter 'y2' (UnusedParameters)

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Style/EmptyLineBetweenDefs: Use empty lines between method definitions.

Comment (yenkins) [inline lib/gooddata/smelly.rb]: Style/SpaceAfterComma: Space missing after comma.

File: lib/gooddata/smelly.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  def foo; end
```
Comment (yenkins) [inline lib/gooddata/smelly.rb]: Style/EmptyLineBetweenDefs: Use empty lines between method definitions.

File: lib/gooddata/smelly.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  def foo!; end
```
Comment (yenkins) [inline lib/gooddata/smelly.rb]: Style/EmptyLineBetweenDefs: Use empty lines between method definitions.

File: lib/gooddata/smelly.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  def bar!; end
```
Comment (yenkins) [inline lib/gooddata/smelly.rb]: Style/EmptyLineBetweenDefs: Use empty lines between method definitions.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/281/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/1158/ : FAILURE in 1m 13s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/307/ : SUCCESS in 47s

**Comment by panjan** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/285/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1/ : FAILURE in 2m 12s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/312/ : SUCCESS in 2m 16s

**Pull request closed without merging**